CMakeLists.txt

```diff
@@ -25,8 +25,26 @@ if(NOT HCC_BIN)
     message(FATAL_ERROR "Can't find hcc")
 endif()
 
+# Support building in new and old device-lib environments
+# Previously installed to /opt/rocm/lib
+# Starting ROCm 3.8 installed to /opt/rocm/amdgcn/bitcode
+find_path(BC_DIR
+  NAMES
+    "opencl.bc"
+    "opencl.amdgcn.bc"
+  PATHS
+    "${ROCM_DIR}/amdgcn/bitcode"
+    "${ROCM_DIR}/lib/bitcode"
+    "${ROCM_DIR}/lib"
+    "${HCC_BIN}/../lib/bitcode"
+    "${HCC_BIN}/../lib"
+    "${CMAKE_INSTALL_PREFIX}/amdgcn/bitcode"
+    "${CMAKE_INSTALL_PREFIX}/lib/bitcode"
+    "${CMAKE_INSTALL_PREFIX}/lib")
```

Comment:
The device libs should be found using `find_package(AMDDeviceLibs PATHS /opt/rocm)` unless we are using hcc which in this case it should use the relative path to hcc.